CMakeLists.txt

```diff
@@ -76,26 +64,32 @@ endif()
 set( CMAKE_EXPORT_COMPILE_COMMANDS ON )
 
 # KS: Compile and link options for more see https://github.com/cpp-best-practices/cppbestpractices/tree/master
-target_compile_options(MaCh3CompilerOptions INTERFACE
-    -g                      # Generate debug information
+add_library(MaCh3Warnings INTERFACE)
+target_compile_options(MaCh3Warnings INTERFACE
     -Wextra                 # Enable extra warning flags
     -Wall                   # Enable all standard warning flags
-    -pedantic               # Enforce strict ISO compliance (all versions of GCC, Clang >= 3.2)
     -Wshadow                # Warn when a variable declaration shadows one from an outer scope
     -Wuninitialized         # Warn about uninitialized variables
     -Wnon-virtual-dtor      # Warn when a class with virtual functions has a non-virtual destructor
     -Woverloaded-virtual    # Warn when a function declaration hides a virtual function from a base class
     -Wformat=2              # Warn on security issues around functions that format output (ie printf)
     -Wunused                # Warn on anything being unused
     -Wredundant-decls       # Warn about multiple declarations of the same entity. Useful for code cleanup.
-    #-Wstrict-aliasing=2    # Helps detect potential aliasing issues that could lead to undefined behavior.
-    #-Wuseless-cast         # Warn if you perform a cast to the same type (only in GCC >= 4.8)
-    #-Wpadded               # Warn when padding is added to a structure or class for alignment
-    #-Wnull-dereference     # Warn if a null dereference is detected (only in GCC >= 6.0)
-    #-Wold-style-cast       # Warn for c-style casts
-    #-Wconversion           # Warn on type conversions that may lose data
-    #-Werror                # Treat Warnings as Errors
+    -Wstrict-aliasing=2     # Helps detect potential aliasing issues that could lead to undefined behavior.
+    -Wuseless-cast          # Warn if you perform a cast to the same type (only in GCC >= 4.8)
+    # -Wpadded               # Warn when padding is added to a structure or class for alignment
+    -Wnull-dereference     # Warn if a null dereference is detected (only in GCC >= 6.0)
+    -Wold-style-cast       # Warn for c-style casts
+    -Wconversion           # Warn on type conversions that may lose data
+    -Werror                # Treat Warnings as Errors
```

Comment:
I can't higlhinth but 
https://github.com/mach3-software/MaCh3/blob/30cb0190dab901768dc5663bc6bedbf362225641/CMakeLists.txt#L95
this problaby should use MaCh3Warnings  instead of MaCh3CompilerOptions

Reply:
yep